exchanges/exchange.go

```diff
@@ -778,52 +783,45 @@ func (e *Base) UpdatePairs(exchangeProducts currency.Pairs, assetType asset.Item
 
 // SetAPIURL sets configuration API URL for an exchange
 func (e *Base) SetAPIURL() error {
-	if e.Config.API.Endpoints.URL == "" || e.Config.API.Endpoints.URLSecondary == "" {
-		return fmt.Errorf("exchange %s: SetAPIURL error. URL vals are empty", e.Name)
-	}
-
-	checkInsecureEndpoint := func(endpoint string) {
-		if strings.Contains(endpoint, "https") {
-			return
+	var err error
+	if e.Config.API.OldEndPoints != nil {
+		if e.Config.API.OldEndPoints.URL != "" && e.Config.API.OldEndPoints.URL != config.APIURLNonDefaultMessage {
+			err = e.API.Endpoints.SetRunning(RestSpot.String(), e.Config.API.OldEndPoints.URL)
+			if err != nil {
+				return err
```

Comment:
I have added in the tests for URLSecondary and WebsocketURL to improve the path coverage. However e.API.Endpoints.SetRunning function returns an error when URL key is invalid. But since we are converting the oldendpoints to new config using hardcoded url keys this error can never be hit. But for the other paths that weren't being covered in the function I've added those in.